src/databricks/labs/ucx/runtime.py

```diff
@@ -434,11 +435,19 @@ def migrate_external_tables_sync(
     table_crawler = TablesCrawler(sql_backend, cfg.inventory_database)
     udf_crawler = UdfsCrawler(sql_backend, cfg.inventory_database)
     grant_crawler = GrantsCrawler(table_crawler, udf_crawler)
-    table_mapping = TableMapping(install, ws, sql_backend)
+    table_mappings = TableMapping(install, ws, sql_backend)
     migration_status_refresher = MigrationStatusRefresher(ws, sql_backend, cfg.inventory_database, table_crawler)
     group_manager = GroupManager(sql_backend, ws, cfg.inventory_database)
+    interactive_grants = PrincipalACL.for_cli(ws, install)
```

Comment:
This is runtime, not CLI.

Construct the object directly, not via the factory. You have most of the dependencies here anyway